src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const headers = table.querySelectorAll('thead th');
+  const tbody = table.querySelector('tbody');
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const rows = Array.from(tbody.querySelectorAll('tr'));
+      const isNumericColumn =
+        header.textContent === 'Age' || header.textContent === 'Salary';
+      const getCellValue = (row) => row.children[index].textContent.trim();
+
+      rows.sort((rowA, rowB) => {
+        const cellA = getCellValue(rowA);
+        const cellB = getCellValue(rowB);
+
+        if (isNumericColumn) {
+          const valueA = parseFloat(cellA.replace(/[^0-9.-]+/g, ''));
+          const valueB = parseFloat(cellB.replace(/[^0-9.-]+/g, ''));
```

Comment:
The use of `parseFloat` to convert cell values to numbers assumes that all values in the column are valid numbers or can be converted to numbers. If a cell contains a non-numeric value or if parsing fails, `parseFloat` will return `NaN`, which can lead to unexpected sorting behavior. Consider adding a check to handle non-numeric values or ensure that the column contains only numeric values.